src/System.Management.Automation/engine/Interop/Windows/CreateHardLink.cs

```diff
@@ -10,7 +10,14 @@ internal static partial class Interop
     internal static unsafe partial class Windows
     {
         [LibraryImport("api-ms-win-core-file-l2-1-0.dll", EntryPoint = "CreateHardLinkW", StringMarshalling = StringMarshalling.Utf16)]
-        [return: MarshalAs(UnmanagedType.Bool)]
-        internal static partial bool CreateHardLink(string name, string existingFileName, nint securityAttributes);
+        private static unsafe partial BOOL CreateHardLink(string lpFileName, string lpExistingFileName, void* lpSecurityAttributes);
```

Comment:
Is there any benefit to using a custom `BOOL` enum here? Dotnet can handle all that for you as it was before. Why use `void*` when `IntPtr` would suffice and remove the need for the unsafe context.

Reply:
I've removed the enum and explained (https://github.com/PowerShell/PowerShell/pull/18707#issuecomment-1335819527) why we shouldn't avoid unsafe context.